From a4d158ce9617cc5adce77b20536523d32ae2d6f6 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Wed, 10 Sep 2014 18:28:43 -0400
Subject: [ethernet] ixgbe: ixgbe calls skb_set_hash

Message-id: <1410373750-11341-24-git-send-email-jogreene@redhat.com>
Patchwork-id: 93542
O-Subject: [RHEL7.1 PATCH 23/50] net: ixgbe calls skb_set_hash
Bugzilla: 1091123
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

BZ: 1091123
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

Drivers should call skb_set_hash to set the hash and its type
in an skbuff.

Signed-off-by: Tom Herbert <therbert@google.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 38da9853aa6d885353f4c96c553ce0462357d5d9)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
index 3b5e626..73d021d 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
@@ -1304,7 +1304,9 @@ static inline void ixgbe_rx_hash(struct ixgbe_ring *ring,
      struct sk_buff *skb)
 {
  if (ring->netdev->features & NETIF_F_RXHASH)
-  skb->rxhash = le32_to_cpu(rx_desc->wb.lower.hi_dword.rss);
+  skb_set_hash(skb,
+        le32_to_cpu(rx_desc->wb.lower.hi_dword.rss),
+        PKT_HASH_TYPE_L3);
 }
 
 #ifdef IXGBE_FCOE
-- 
1.7.1